### PR TITLE
Enable building calendar for a particular seder

### PR DESCRIPTION
**Pull request by kalmanh**
Thought this would be neat. Let me know what you think.

תפסת מרובה לא תפסת. I am one of those guys that likes setting small goals. :)

For now, this will only work for seder nashim - it won't take me long to add others, if you think there is some merit to this PR.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,6 +16,11 @@ var total_m = _.reduce(flatArray, function(memo, num){ return memo + parseInt(nu
 var names = _.keys(json);
 var hebrew = JSON.parse(fs.readFileSync(path.join(__dirname, "data", "hebrew.json")));
 
+// Sedarim
+var seder = {};
+seder.nashim = ["Yevamot", "Ketubot", "Nedarim", "Nazir", "Sotah", "Gittin", "Kidushin"];
```
Comment (dbankier) [inline index.js]: The sedarim could/should be managed by the config files in `/data`.

File: index.js [hunk 2]
```diff
@@ -85,13 +90,36 @@ exports.getToday = function (_date, pretty) {
...
+
+  var allowedMasechtos = _.map(seder[thisSeder], function(masechtaName){ return _.indexOf(names, masechtaName) });
+
+  var partOfSeder = function(dailyQuota) { return _.contains(allowedMasechtos,  dailyQuota.t); }
+  list = _.chain(list)
+    .filter(function(dailyQuota){
+      return _.some(dailyQuota, partOfSeder);
+    }).map(function(dailyQuota){
+      return _.filter(dailyQuota, partOfSeder);
+    }).value();
```
Comment (dbankier) [inline index.js]: A simple `reduce` function could do this instead of using underscore.

File: index.js [hunk 3]
```diff
@@ -85,13 +90,36 @@ exports.getToday = function (_date, pretty) {
...
 exports.buildCalendar = function (o) {
   mas_index = 0;
   perek_index = 0;
   mish_index = 0;
 
   var day = new moment(o.start || moment([2010, 6, 4]));
   var list = buildList(o.per_day || 2, o.perakim);
+
+  if(o.seder){
+    list = constrainListToSeder(list, o.seder);
```
Comment (dbankier) [inline index.js]: Also you might want to do the limiting in the buildList function (more efficient).

**Comment by kalmanh** [comment]
@dbankier - I addressed your concerns in the new commit.

Not sure what you meant by 

> A simple reduce function could do this instead of using underscore.

although I guess that's [OBE](http://www.acronymfinder.com/Overcome-By-Events-%28OBE%29.html) now.

If you can point me to what this "reduce function" thing is - I would appreciate it. I am more of a Java/Ruby/Rails guy - slowly getting over my aversion to JS with Node.js :)

**Comment by dbankier** [comment]
Thank you

**Pull request merged**